lib/client/index.js

```diff
@@ -192,6 +192,67 @@ class Client {
     return () => this._supscriptions.delete(url, callback)
   }
 
+  /**
+   * Returns a list of items in a directory.
+   *
+   * @param {string} directory
+   * @param {object} [opts]
+   * @param {boolean} [opts.skipCache] - Skip the local cache and wait for the remote relay to respond with fresh data
+   *
+   * @returns {Promise<Array<string>>}
+   */
+  async list (directory, opts = {}) {
+    if (!directory.endsWith('/')) directory += '/'
+    if (!directory.startsWith('/')) directory = '/' + directory
+
+    if (opts.skipCache) {
+      return this._listRelay(directory)
+    }
+
+    return this._listLocal(directory)
+  }
+
+  /**
+   * Returns a list of items in a directory from local store.
+   *
+   * @param {string} directory
+   *
+   * @returns {Promise<Array<string>>}
+   */
+  async _listLocal (directory) {
+    const range = {
+      gte: PREFIXES.RECORDS + this.id + directory,
+      //
+      lte: PREFIXES.RECORDS + this.id + directory.slice(0, directory.length - 1) + '~'
+    }
+
+    const list = await this._store.iterator(range).all()
+
+    return list.map(r => r[0].split(PREFIXES.RECORDS + this.id + directory)[1])
```

Comment:
Recommend that we don't strip the subdir here so that clients can query parent paths, ie.:
```js
return list.map(r => r[0].split(PREFIXES.RECORDS + this.id)[1])
```

Reply:
I added a `client.query()` method for that, which `client.list()` is using under the hood.